### DISCUSSION TITLE
Asking a wecsim capability

### DISCUSSION THREAD

**Discussion by HareendraAmaratunge** [WEC-Sim/WEC-Sim#1326] [Q&A] [answered]
I want to know, by utilizing this software in the Matlab platform, Am I be able to get the energy of a wave which we defined in the wave information part of the wecsiminput file?

**Comment by dav-og** [comment, marked as answer]
You might want to look into the plotSpectrum() function in WEC-Sim's waveClass:

https://wec-sim.github.io/WEC-Sim/main/user/api.html#objects.waveClass.plotSpectrum

**Comment by HareendraAmaratunge** [comment]
Thank you for being kind enough to reply. I need to know how I can plot the
wave energy spectrum. I didn't answered for that question. I went to the
link you shared and it doesn't help me to write the code to generate the
wave energy spectrum. Please help me.

Thanks

**Comment by kmruehl** [comment]
@HareendraAmaratunge the link @dav-og provided is to the API documentation for the `waves.plotSpectrum` method, which plots the wave energy spectrum. The source code for this method is [here](https://github.com/WEC-Sim/WEC-Sim/blob/8177e1cf593616c3ec4854d81d319adb402897f8/source/objects/waveClass.m#L499).

**Comment by HareendraAmaratunge** [comment]
Hii,
I tried the source code supplied. and I came up with this. What Can I do
now?
